msteams-platform/tabs/how-to/dashboard-teams-toolkit.md

```diff
@@ -0,0 +1,991 @@
+
+Example of a dashboard:
+
+:::image type="content" source="../../assets/images/dashboard/dashboard-demonstration.png" alt-text="Screenshot shows the sample of a dashboard.":::
+
+**Advantages**
+
+* Dashboards connect numerous metrics, data sources, APIs, and services to help business extract relevant information from the sources and present it to the users.
+
+* Customizable dashboards allow your business to set specific goals that help you track the information you need to view in multiple areas and across departments.
+
+## Add a new dashboard
+
+Following are the steps to add a new dashboard layout:
+
+1. [Create a dashboard class](#create-a-dashboard-class)
+1. [Override methods to customize dashboard layout](#override-methods-to-customize-dashboard-layout)
+1. [Add a route for the new dashboard](#add-a-route-for-the-new-dashboard)
+1. [Modify manifest to add a new dashboard tab](#modify-manifest-to-add-a-new-dashboard-tab)
+
+### Create a dashboard class
+
+Create a file with the extension `.tsx` for your dashboard in the tabs/src/views/dashboards directory, for example, YourDashboard.tsx. Then, create a class that extends the Dashboard class:
+
+```typescript
+
+export default class YourDashboard extends Dashboard {}
+
+```
+
+export default class YourDashboard extends Dashboard {}
+
+### Override methods to customize dashboard layout
+
+Dashboard class provides few methods that you can override to customize the dashboard layout. The following table lists the methods that you can override:
+
+| **Methods** | **Function** |
+|---|---|
+| `rowHeight()` | Customize the height of each row of the dashboard. |
+| `columnWidth()` | Customize how many columns the dashboard has at most and the width of each column. |
+| `dashboardLayout()` | Define widgets layout. |
+
+Following is an example to customize the dashboard layout.
+
+```typescript
+export default class YourDashboard extends Dashboard {
+  protected rowHeights(): string | undefined {
+    return "500px";
+  }
+
+  protected columnWidths(): string | undefined {
+    return "4fr 6fr";
+  }
+
+  protected dashboardLayout(): void | JSX.Element {
+    return (
+      <>
+        <SampleWidget />
+        <div style={oneColumn("6fr 4fr")}>
+          <SampleWidget />
+          <SampleWidget />
+        </div>
+      </>
+    );
+  }
+}
+```
+
+>[!NOTE]
+> All methods are optional. If you don't override any method, the default dashboard layout will be used.
+
+### Add a route for the new dashboard
+
+Open tabs/src/App.tsx file and add a route for the new dashboard. Here's an example:
+
+```typescript
+import YourDashboard from "./views/dashboards/YourDashboard";
+
+export default function App() {
+  ...
+  <Route exact path="/yourdashboard" component={YourDashboard} />
+  ...
+}
+```
+
+### Modify manifest to add a new dashboard tab
+
+Open templates/appPackage/manifest.template.json file and add a new dashboard tab under the `staticTabs`. Here's an example:
+
+```typescript
```

Comment:
The language of this code snippet should be JSON.

Reply:
Updated.